core/graphelier-service/models/orderbooks.go

```diff
@@ -267,3 +267,29 @@ func (orderbook *Orderbook) ApplyMessagesToDeltabook(messages []*Message, numMes
 
 	return
 }
+
+// TopBookPerXNano : Returns a list of the top book at every x nanosecond after applying each individual message in a list
+func (orderbook *Orderbook) TopBookPerXNano(messages []*Message, x uint64) (topbook []*Point) {
+	topbookLengthIndex := uint64(0)
+	for messagesIndex, message := range messages {
```

Comment:
As @chimano had mentioned earlier, you are going to need more logic to make sure you are creating points at regular intervals and not just when there is a message. For example you do not generate points between the start of the interval and the first message, between the last message and the end of the interval, and you are only generating at most 1 point between 2 messages while there could be many.